mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -1,32 +1,55 @@
-use super::{broadcast_p2p_tx_msg, lp_coinfind, tx_helper_topic, wait_for_taker_payment_conf_until, H256Json,
-            SwapsContext, TransactionIdentifier, WAIT_CONFIRM_INTERVAL};
+use super::{broadcast_p2p_tx_msg, lp_coinfind, taker_payment_spend_deadline, tx_helper_topic, H256Json, SwapsContext,
+            WAIT_CONFIRM_INTERVAL};
 use crate::mm2::MmError;
 use async_trait::async_trait;
-use coins::{CanRefundHtlc, MmCoinEnum, WatcherValidatePaymentInput};
+use coins::{CanRefundHtlc, FoundSwapTxSpend, MmCoinEnum, WatcherSearchForSwapTxSpendInput,
+            WatcherValidatePaymentInput, WatcherValidateTakerFeeInput};
 use common::executor::{AbortSettings, SpawnAbortable, Timer};
-use common::log::{error, info};
-use common::now_ms;
+use common::log::{debug, error, info};
 use common::state_machine::prelude::*;
+use common::{now_ms, DEX_FEE_ADDR_RAW_PUBKEY};
 use futures::compat::Future01CompatExt;
 use mm2_core::mm_ctx::MmArc;
 use mm2_libp2p::{decode_signed, pub_sub_topic, TopicPrefix};
-use mm2_number::BigDecimal;
-use rpc::v1::types::Bytes as BytesJson;
 use std::cmp::min;
 use std::sync::Arc;
 use uuid::Uuid;
 
 pub const WATCHER_PREFIX: TopicPrefix = "swpwtchr";
 const TAKER_SWAP_CONFIRMATIONS: u64 = 1;
 pub const TAKER_SWAP_ENTRY_TIMEOUT: u64 = 21600;
-const WAIT_FOR_TAKER_REFUND: u64 = 1200; // How long?
 
 struct WatcherContext {
     ctx: MmArc,
     taker_coin: MmCoinEnum,
     maker_coin: MmCoinEnum,
-    data: TakerSwapWatcherData,
     verified_pub: Vec<u8>,
+    data: TakerSwapWatcherData,
+}
+
+impl WatcherContext {
+    fn taker_locktime(&self) -> u64 { self.data.swap_started_at + self.data.lock_duration }
+
+    fn wait_for_maker_payment_spend_deadline(&self) -> u64 {
+        match std::env::var("SKIP_WAIT_FOR_MAKER_PAYMENT_SPEND") {
```

Comment:
It turned out that there are too many of such envs 🙂 Could you please consider moving these to a new `watcher_conf` object in JSON conf? I suppose that it might be nice to have a possibility to change these and play with different values during testing.